core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -532,11 +625,23 @@ class LogTest extends JUnitSuite {
       bw.close()
     }
 
+    // corrupt all the index files
+    for( file <- timeIndexFiles) {
+      val bw = new BufferedWriter(new FileWriter(file))
+      bw.write("  ")
+      bw.close()
+    }
+
     // reopen the log
     log = new Log(logDir, config, recoveryPoint = 200L, time.scheduler, time)
```

Comment:
In our case, we will have corrupted log index, so no matter what recovery point it is, the index will be deleted and rebuilt.